### PR TITLE
Gamepad stuffs

### PR DESCRIPTION
**Pull request by WALLOFJUSTICE**

### CODE REVIEW

File: lang/en.txt [hunk 1]
```diff
@@ -6040,5 +6040,10 @@ Cast#
 4086 View item details#
 4087 Hide item details#
 4088 You found %d gold pieces in the chest!#
+# Nav Prompts
+4089 Move cursor#
+4093 View Spells#
+4094 View Items#
+4095 View Skills#
```
Comment (SheridanR) [inline lang/en.txt:6047]: I used a couple of these for crab-related messages. Just a heads up that these will have to change when we merge

File: src/actplayer.cpp [hunk 2]
```diff
@@ -2859,6 +2859,11 @@ void actPlayer(Entity* my)
 		    }
 		    messagePlayer(PLAYER_NUM, MESSAGE_DEBUG, "Removed keyboard for any player");
 	    }
+		if ( keystatus[SDL_SCANCODE_LCTRL] && keystatus[SDL_SCANCODE_KP_7] )
```
Comment (SheridanR) [inline src/actplayer.cpp:2862]: can you gate this behind `enableDebugKeys`

File: src/game.cpp [hunk 3]
```diff
@@ -5865,9 +5871,24 @@ int main(int argc, char** argv)
 					{
 						continue;
 					}
-					if ((subwindow && !players[i]->shootmode) || (gamePaused && i == clientnum))
+
+					if ( gamePaused )
```
Comment (SheridanR) [inline src/game.cpp:5875]: this block doesn't compile in editor, add `#ifndef EDITOR` here

File: src/ui/Button.cpp [hunk 4]
```diff
@@ -75,7 +75,19 @@ void Button::draw(SDL_Rect _size, SDL_Rect _actualSize, const std::vector<const
 #ifdef EDITOR
 	bool focused = highlighted || selected;
 #else
-	int mouseowner = intro ? 0 : owner;
+	int mouseowner_pausemenu = 0;
+	if ( gamePaused )
```
Comment (SheridanR) [inline src/ui/Button.cpp:79]: as with Button.cpp this block needs an `#ifndef EDITOR`

File: src/ui/Button.cpp [hunk 5]
```diff
@@ -348,7 +360,19 @@ Button::result_t Button::process(SDL_Rect _size, SDL_Rect _actualSize, const boo
 		return result;
 	}
 
-	int mouseowner = intro ? 0 : owner;
+	int mouseowner_pausemenu = 0;
+	if ( gamePaused )
```
Comment (SheridanR) [inline src/ui/Button.cpp:364]: as with Button.cpp this block needs an `#ifndef EDITOR`

File: src/ui/Field.cpp [hunk 6]
```diff
@@ -354,7 +354,19 @@ Field::result_t Field::process(SDL_Rect _size, SDL_Rect _actualSize, const bool
 		return result;
 	}
 
-	int mouseowner = intro ? 0 : owner;
+	int mouseowner_pausemenu = 0;
+	if ( gamePaused )
```
Comment (SheridanR) [inline src/ui/Field.cpp:358]: as with Button.cpp this block needs an `#ifndef EDITOR`

File: src/ui/Frame.cpp [hunk 7]
```diff
@@ -345,7 +345,19 @@ void Frame::draw(SDL_Rect _size, SDL_Rect _actualSize, const std::vector<const W
 		}
 	}
 
-	int mouseowner = intro ? 0 : owner;
+	int mouseowner_pausemenu = 0;
+	if ( gamePaused )
```
Comment (SheridanR) [inline src/ui/Frame.cpp:349]: as with Button.cpp this block needs an `#ifndef EDITOR`

File: src/ui/Frame.cpp [hunk 8]
```diff
@@ -677,7 +689,19 @@ Frame::result_t Frame::process(SDL_Rect _size, SDL_Rect _actualSize, const std::
 	fullSize.h += (actualSize.w > size.w) ? sliderSize : 0;
 	fullSize.w += (actualSize.h > size.h) ? sliderSize : 0;
 
-	int mouseowner = intro ? 0 : owner;
+	int mouseowner_pausemenu = 0;
+	if ( gamePaused )
```
Comment (SheridanR) [inline src/ui/Frame.cpp:693]: as with Button.cpp this block needs an `#ifndef EDITOR`

File: src/ui/Slider.cpp [hunk 9]
```diff
@@ -26,7 +26,19 @@ void Slider::draw(SDL_Rect _size, SDL_Rect _actualSize, const std::vector<const
 #ifdef EDITOR
 	bool focused = highlighted || selected;
 #else
-	int mouseowner = intro ? 0 : owner;
+	int mouseowner_pausemenu = 0;
+	if ( gamePaused )
```
Comment (SheridanR) [inline src/ui/Slider.cpp:30]: as with Button.cpp this block needs an `#ifndef EDITOR`

File: src/ui/Slider.cpp [hunk 10]
```diff
@@ -215,7 +227,19 @@ Slider::result_t Slider::process(SDL_Rect _size, SDL_Rect _actualSize, const boo
 		return result;
 	}
 
-	int mouseowner = intro ? 0 : owner;
+	int mouseowner_pausemenu = 0;
+	if ( gamePaused )
```
Comment (SheridanR) [inline src/ui/Slider.cpp:231]: as with Button.cpp this block needs an `#ifndef EDITOR`

**Review by SheridanR — CHANGES_REQUESTED** [review]

File: src/actplayer.cpp (hunk 2 above)
Comment (WALLOFJUSTICE) [reply src/actplayer.cpp:2862]: The block around these keys are already gated with `enableDebugKeys`

File: src/game.cpp (hunk 3 above)
Comment (WALLOFJUSTICE) [reply src/game.cpp:5875]: Don't believe game.cpp is linked by editor, so no issues here

File: lang/en.txt (hunk 1 above)
Comment (WALLOFJUSTICE) [reply lang/en.txt:6047]: Moved language[4089] to [4092], crab messages 4089,4090 should be fine now, just need to combine these files at merge time

**Pull request merged**